ast/zson.go

```diff
@@ -0,0 +1,174 @@
+package ast
+
+type Value interface {
+	valueNode()
+}
+
+const (
+	ImpliedValueOp = "implied_value"
+	CastValueOp    = "cast_value"
+	DefValueOp     = "def_value"
+)
+
+type ImpliedValue struct {
+	Op string `json:"op"`
+	Of Any    `json:"of"`
+}
+
+type DefValue struct {
+	Op       string `json:"op"`
+	Of       Any    `json:"of"`
+	TypeName string `json:"type_name"`
+}
+
+type CastValue struct {
+	Op   string `json:"op"`
+	Of   Value  `json:"of"`
+	Type Type   `json:"type"`
+}
+
+func (*ImpliedValue) valueNode() {}
+func (*DefValue) valueNode()     {}
+func (*CastValue) valueNode()    {}
+
+type Any interface {
+	anyNode()
+}
+
+const (
+	PrimitiveOp = "primitive"
+	RecordOp    = "record"
+	ArrayOp     = "array"
+	SetOp       = "set"
+	UnionOp     = "union"
+	EnumOp      = "enum"
+	MapOp       = "map"
+	NullOp      = "null"
+	TypeValueOp = "type_value"
+)
+
+type (
+	Primitive struct {
+		Op   string `json:"op"`
+		Type string `json:"type"`
+		Text string `json:"text"`
+	}
+	Record struct {
+		Op     string  `json:"op"`
+		Fields []Field `json:"fields"`
+	}
+	Field struct {
+		Name  string `json:"name"`
+		Value Value  `json:"value"`
+	}
+	Array struct {
+		Op       string  `json:"op"`
+		Elements []Value `json:"elements"`
+	}
+	Set struct {
+		Op       string  `json:"op"`
+		Elements []Value `json:"elements"`
+	}
+	Enum struct {
+		Op   string `json:"op"`
+		Name string `json:"name"`
+	}
+	Map struct {
+		Op      string  `json:"op"`
+		Entries []Entry `json:"entries"`
+	}
+	Entry struct {
+		Key   Value `json:"key"`
+		Value Value `json:"value"`
+	}
+	TypeValue struct {
+		Op   string `json:"op"`
+		Type Type   `json:"type"`
```

Comment:
maybe: 

```suggestion
		Value Type   `json:"value"`
```

(I came to this after looking at some zson ast while working on reshape...)

Reply:
Yes!  We're getting the hang of this.  I had it right in zson.TypeValue.